drivers/lte_link_control/lte_lc.c

```diff
@@ -68,13 +67,22 @@ static const char legacy_pco[] = "AT%XEPCO=0";
 
 void at_handler(char *response)
 {
+	char  id[16];
+	u32_t val;
+	size_t len = 16;
+
 	LOG_DBG("recv: %s", log_strdup(response));
 
-	if (!memcmp(status1, response, AT_CMD_SIZE(status1)) ||
-	    !memcmp(status2, response, AT_CMD_SIZE(status2)) ||
-	    !memcmp(status3, response, AT_CMD_SIZE(status3)) ||
-	    !memcmp(status4, response, AT_CMD_SIZE(status4))) {
-		k_sem_give(&link);
+	at_parser_params_from_str(response, NULL, &params);
+	at_params_string_get(&params, 0, id, &len);
+
+	if ((len > 0) &&
```

Comment:
You could add a short explanation here to explain the AT response you are waiting for.

Reply:
Ok

Reply:
Nit: typo, remove one "from".